react-native-aztec/ios/RNTAztecView/RCTAztecView.swift

```diff
@@ -178,26 +178,41 @@ class RCTAztecView: Aztec.TextView {
     }
     
     // MARK: - Paste handling
-    
-    private func html(from pasteboard: UIPasteboard) -> String? {
-        guard let data = pasteboard.data(forPasteboardType: kUTTypeHTML as String) else {
-            return nil
+    private func read(from pasteboard: UIPasteboard, uti: String, documentType: DocumentType) -> String? {
+        guard let data = pasteboard.data(forPasteboardType: uti as String),
+            let attributedString = try? NSAttributedString(data: data, options: [.documentType: documentType], documentAttributes: nil),
+            let storage = self.textStorage as? TextStorage else {
+                return nil
         }
-        
-        return String(data: data, encoding: .utf8)
+        return  storage.getHTML(from: attributedString)
     }
-    
-    private func text(from pasteboard: UIPasteboard) -> String? {
-        guard let data = pasteboard.data(forPasteboardType: kUTTypePlainText as String) else {
-            return nil
+
+    private func readHTML(from pasteboard: UIPasteboard) -> String? {
+
+        if let data = pasteboard.data(forPasteboardType: kUTTypeHTML as String), let html = String(data: data, encoding: .utf8) {
+            // Make sure we are not getting a full HTML DOC. We only want inner content
+            if !html.hasPrefix("<!DOCTYPE html") {
```

Comment:
I'm doing this check here to check if we are not importing a full HTML document because on my tests I found that the when copying from the Notes app, it was sending the full HTML document, using `span` and styles for bold/italic styles instead of using just `<strong>` and `<em>` element. 

My assumption is that when importing HTML if a full document is there, maybe there is content there is not valid or importable to GB so instead we should try to read RTF.

If think that this is not a good assumption we can remove this line, but then import from the Notes app will not work.

Reply:
> using span and styles for bold/italic styles instead of using just <strong> and <em> element.
> 
> My assumption is that when importing HTML if a full document is there, maybe there is content there is not valid or importable to GB so instead we should try to read RTF.

FWIW, [code exists on GB side to handle inline styles on span tags](https://github.com/WordPress/gutenberg/blob/master/packages/blocks/src/api/raw-handling/phrasing-content-reducer.js#L9), however, the limitations of our DOM implementation (jsdom-jscore) prevent it from working correctly.

To "enable" this code path, we'd need the style property on `ELEMENT_NODE`s to be implemented, as well as `ATTRIBUTE_NODE`s.

> If we think that this is not a good assumption we can remove this line, but then import from the Notes app will not work.

It's possible that implementing the necessary DOM spec will require significant time / effort. I think your approach is good to mitigate the underlying problem (that our DOM implementation is insufficient). I wonder if wrapping the html in a "full document" is correlated with using inline styles on spans, or if Notes.app happens to behave this way, and other apps have differing behavior regarding these two characteristics.

Is there some way to directly check if the pasteboard html generated by the external app uses inline styles on span tags?

Reply:
>  I think your approach is good to mitigate the underlying problem

I agree with this!

>  I wonder if wrapping the html in a "full document" is correlated with using inline styles on spans, or if Notes.app happens to behave this way, and other apps have differing behavior regarding these two characteristics.

I just tested the Google docs app, and it happens that it does use inline styles on spans, but without wrapping it up on a full HTML document. So this approach doesn't work for it.

But it does work for the Notes app.

I'm OK with shipping this improvement to support the Apple default's Note app, and check for a more wide alternative later on.

What do you think @SergioEstevao ?